draft-ietf-ace-oscore-profile.xml

```diff
@@ -141,15 +141,19 @@
       </t>
 
       <t>
-        Once the client has retrieved the access token, it posts it to the RS using the authz-info endpoint and mechanisms specified in section 5.8 of <xref target="I-D.ietf-ace-oauth-authz"/>. If the access token is valid, the RS replies to this request with a 2.01 (Created) response, which contains a nonce N1.
+        Once the client has retrieved the access token, it generates a nonce N1 and posts both the token and N1 to the RS using the authz-info endpoint and mechanisms specified in section 5.8 of <xref target="I-D.ietf-ace-oauth-authz"/> and Content-Format = application/ace+cbor.
       </t>
 
       <t>
-        After receiving the nonce N1, the client generates a nonce N2, concatenates it with N1 and sets the ID Context in its Security Context (see section 3 of <xref target="I-D.ietf-core-object-security"/>) to N1 concatenated with N2. The client then derives the complete Security Context from the ID Context plus the parameters received from the AS.
+        If the access token is valid, the RS replies to this request with a 2.01 (Created) response, which contains a nonce N2. Moreover, the server concatenates N1 with N2 and sets the ID Context in the Security Context (see section 3 of <xref target="I-D.ietf-core-object-security"/>) to N1 concatenated with N2. The RS then derives the complete Security Context associated with the received token from it plus the parameters received in the AS, following section 3.2 of <xref target="I-D.ietf-core-object-security"/>.
       </t>
```

Comment:
No, what I am saying is I would like the server to be able to tell the client what the sender ID it should be using is.  That is it can make sure there are no collisions when it is lookup up the ID on an incoming message.  I am not sure that we should be doing this it is just a thought.

Reply:
Ah ok, so it is the ClientId that would be chosen by the RS. Yes, that is also doable, it would require a bit of additional processing described (can we make the ClientId in the OSCORE_Security_Context optional? if it is, then we need to specify that if the RS does not receive a ClientId then it MUST send a ClientId back. Alternatively, the RS MAY send a ClientId back, in which case the Client MUST use it) but apart from that, I don't think it would open up to attacks. 2.01 response is not protected, so an attacker could change (or inject) a ClientId, but it could also change or inject the nonce, and that would have the same effect (different Sec Ctx created on client and server). I will give it a try now and add it to the text.

Reply:
Done now here: ae4d98d